test/integration/client/client_test.go

```diff
@@ -1366,12 +1366,12 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 			WithTemplate(corev1ac.PodTemplateSpec().
 				WithSpec(corev1ac.PodSpec().
 					WithContainers(
-						corev1ac.Container().
-							WithName("nginx").
-							WithWorkingDir("/tmp/v2"),
 						corev1ac.Container().
 							WithName("sidecar").
 							WithImage("nginx:1.14.3"),
+						corev1ac.Container().
+							WithName("nginx").
+							WithWorkingDir("/tmp/v2"),
```

Comment:
previously, applying a non-overlapping item in a list appended, and now it prepends? that's surprising... I thought the change was only modifying preferring the request order when it specified items already present

Reply:
I missed that in the master PR... let's resolve this question in master before merging the cherry picks

Reply:
This is a side effect of preferring the RHS order. The implementation would append everything without defined ordering to the result. Before the change, everything in LHS has defined order and thus each items in disjointed RHS would be appended. Now the situation reversed between LHS and RHS. Everything in disjointed LHS would be appended instead.

Reply:
I don't think the new behavior is correct either.

This is what happens now:
* apply `a,b,c` as fieldManager `x`
* apply `c,d,e,f` as fieldManager `y`
* final order is `c,d,e,f,a,b`

That discards the order expressed by fieldManager `x`, which is unexpected

Reply:
moving discussion to the original issue at https://github.com/kubernetes/kubernetes/issues/104641#issuecomment-976947921